{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import pingouin as pg\n",
    "from scipy.stats import spearmanr\n",
    "from sklearn.metrics import confusion_matrix\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "import quilt3\n",
    "\n",
    "from fish_morphology_code.analysis.notebook_utils import (\n",
    "    DAY_18_COLOR,\n",
    "    DAY_32_COLOR,\n",
    "    DAY_COLOR_PALETTE,\n",
    "    DAY_COLOR_PALETTE_THREE,\n",
    "    BAR_PLOT_COLUMNS,\n",
    "    SHORT_FEAT_NAME_MAP,\n",
    "    BAR_PLOT_COLUMNS_SHORT,\n",
    "    PROBE_ORDER,\n",
    "    CI_EXTENT,\n",
    "    FEATURE_TYPE_MAP,\n",
    "    safe,\n",
    "    get_regression_coef,\n",
    "    boot_regression,\n",
    "    ci_low,\n",
    "    ci_high,\n",
    "    make_reg_plot_ci_df,\n",
    "    make_regression_bar_plot,\n",
    "    get_pred_true,\n",
    "    make_regression_scatter_plot,\n",
    "    boot_spearmanr,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## plot config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "SAVE=False\n",
    "\n",
    "if SAVE:\n",
    "    SAVE_DIR = Path(\"./plots\")\n",
    "    SAVE_DIR.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "    save_dir_pngs = SAVE_DIR/\"pngs\"\n",
    "    save_dir_svgs = SAVE_DIR/\"svgs\"\n",
    "    save_dir_pngs.mkdir(parents=True, exist_ok=True)\n",
    "    save_dir_svgs.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "    fig4_supp_dir_pngs = save_dir_pngs/\"fig4_supp\"\n",
    "    fig4_supp_dir_svgs = save_dir_svgs/\"fig4_supp\"\n",
    "    fig4_supp_dir_pngs.mkdir(parents=True, exist_ok=True)\n",
    "    fig4_supp_dir_svgs.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "    regression_sup_dir_pngs = save_dir_pngs/\"regression_supp\"\n",
    "    regression_sup_dir_svgs = save_dir_svgs/\"regression_supp\"\n",
    "    regression_sup_dir_pngs.mkdir(parents=True, exist_ok=True)\n",
    "    regression_sup_dir_svgs.mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p = quilt3.Package.browse(\n",
    "    \"aics/integrated_transcriptomics_structural_organization_hipsc_cm\",\n",
    "    \"s3://allencell\",\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FISH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = p[\"manuscript_plots\"][\"data.csv\"]()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### scRNA-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_rnaseq = p[\"manuscript_plots\"][\"data_rnaseq.csv\"]()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make tidy version of FISH data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "melt_feats = {\n",
    "    \"count\" : [\n",
    "        'HPRT1 count',\n",
    "        'COL2A1 count',\n",
    "        'H19 count',\n",
    "        'ATP2A2 count',\n",
    "        'MYH6 count',\n",
    "        'MYH7 count',\n",
    "        'BAG3 count',\n",
    "        'TCAP count',\n",
    "    ],\n",
    "    \"(count/μm^2)\" : [\n",
    "        'HPRT1 (count/μm^2)',\n",
    "        'COL2A1 (count/μm^2)',\n",
    "        'H19 (count/μm^2)',\n",
    "        'ATP2A2 (count/μm^2)',\n",
    "        'MYH6 (count/μm^2)',\n",
    "        'MYH7 (count/μm^2)',\n",
    "        'BAG3 (count/μm^2)',\n",
    "        'TCAP (count/μm^2)'\n",
    "    ]\n",
    "}\n",
    "\n",
    "melted_dfs = {\n",
    "    feat:df.melt(\n",
    "        id_vars=[c for c in df.columns if c not in cols],\n",
    "        value_vars=cols,\n",
    "        var_name=\"FISH probe\",\n",
    "        value_name=f\"FISH probe {feat}\",\n",
    "    ).dropna(\n",
    "        subset=[f\"FISH probe {feat}\"]\n",
    "    ).reset_index(\n",
    "        drop=True\n",
    "    )\n",
    "    for feat, cols in melt_feats.items()\n",
    "}\n",
    "\n",
    "for feat,v in melted_dfs.items():\n",
    "    melted_dfs[feat][\"FISH probe\"] = melted_dfs[feat][\"FISH probe\"].str.split().str.get(0)\n",
    "    \n",
    "df_tidy = melted_dfs['count'].merge(\n",
    "    melted_dfs[\"(count/μm^2)\"]\n",
    ").drop(\n",
    "    columns=melt_feats['count']+melt_feats[\"(count/μm^2)\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make version that counts both human scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "melt_cols = [\n",
    "        'Expert structural annotation score (annotator 1)',\n",
    "        'Expert structural annotation score (annotator 2)',\n",
    "    ]\n",
    "\n",
    "keep_cols = [\n",
    "        'Cell number',\n",
    "        'FOV path',\n",
    "        'Cell age',\n",
    "        'Combined organizational score',\n",
    "        \"Combined organizational score (rounded)\",\n",
    "        'Alpha-actinin intensity (median, normalized per day)',\n",
    "    ]\n",
    "\n",
    "df_org_scores_all = df[melt_cols+keep_cols].melt(\n",
    "    id_vars=keep_cols,\n",
    "    value_vars=melt_cols,\n",
    "    var_name=\"Score origin\",\n",
    "    value_name=\"Expert structural annotation score (all)\",\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tidy.groupby(\n",
    "    [\"Cell age\", \"Expert structural annotation score (grouped)\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(df[['Cell number', 'FOV path']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(df[['FOV path']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.groupby(\n",
    "    [\"Cell age\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tidy.groupby(\n",
    "    [\"FISH probe\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tidy.groupby(\n",
    "    [\"FISH probe\", \"Cell age\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tidy[[\"FOV path\", \"FISH probe\"]].drop_duplicates().groupby(\n",
    "    [\"FISH probe\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(4, 4))\n",
    "\n",
    "plot_67_seq = sns.scatterplot(\n",
    "    data=df_rnaseq,\n",
    "    x=\"MYH6 expression\",\n",
    "    y=\"MYH7 expression\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"12\", \"24\"],\n",
    "    palette=day_color_palette,\n",
    "    linewidth=0,\n",
    "    alpha = 0.5,\n",
    "    s=10\n",
    ")\n",
    "\n",
    "plot_67_seq.set(\n",
    "    xlim=(0, 6),\n",
    "    ylim=(0, 6),\n",
    "    xlabel=\"MYH6 transcript abundance\",\n",
    "    ylabel=\"MYH7 transcript abundance\",\n",
    ")\n",
    "\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.7), frameon=False)\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "plt.savefig(save_dir_pngs/'fig_2c_rnaseq_scatter.png', dpi=100, bbox_inches = \"tight\")\n",
    "plt.savefig(save_dir_svgs/'fig_2c_rnaseq_scatter.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df_rnaseq,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"12\", \"24\"],\n",
    "    palette=day_color_palette,\n",
    "    height=1.5,\n",
    "    aspect=2.33,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"MYH6 expression\",\n",
    "    bins=bins,\n",
    "    kde_kws=dict(bw=0.1)\n",
    ")\n",
    "g.set(xlabel='', ylabel='Probability density')\n",
    "g.set(xlim=(0.3, 6));\n",
    "g.set(ylim=(0, 0.6));\n",
    "\n",
    "# save png and svg\n",
    "plt.savefig(save_dir_pngs/'fig_2c_rnaseq_MYH6_hist_main.png', dpi=100, bbox_inches = \"tight\")\n",
    "plt.savefig(save_dir_svgs/'fig_2c_rnaseq_MYH6_hist_main.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df_rnaseq,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"12\", \"24\"],\n",
    "    palette=day_color_palette,\n",
    "    height=1.5,\n",
    "    aspect=0.3,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"MYH6 expression\",\n",
    "    bins=bins,\n",
    "    kde=False,\n",
    "    norm_hist=True,\n",
    ")\n",
    "g.set(xlabel='', ylabel='Probability density')\n",
    "g.set(xlim=(-0.3, 0.3));\n",
    "g.set(ylim=(0, None));\n",
    "g.set(xticklabels=[]);\n",
    "\n",
    "# save png and svg\n",
    "plt.savefig(save_dir_pngs/'fig_2c_rnaseq_MYH6_hist_zeros.png', dpi=100, bbox_inches = \"tight\")\n",
    "plt.savefig(save_dir_svgs/'fig_2c_rnaseq_MYH6_hist_zeros.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df_rnaseq,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"12\", \"24\"],\n",
    "    palette=day_color_palette,\n",
    "    height=1.5,\n",
    "    aspect=2.33,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"MYH7 expression\",\n",
    "    bins=bins,\n",
    "    kde_kws=dict(bw=.1),\n",
    ")\n",
    "g.set(xlabel='', ylabel='Probability density')\n",
    "g.set(xlim=(0.3, 6));\n",
    "g.set(ylim=(0, 0.4));\n",
    "\n",
    "# save png and svg\n",
    "plt.savefig(save_dir_pngs/'fig_2c_rnaseq_MYH7_hist_main.png', dpi=100, bbox_inches = \"tight\")\n",
    "plt.savefig(save_dir_svgs/'fig_2c_rnaseq_MYH7_hist_main.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df_rnaseq,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"12\", \"24\"],\n",
    "    palette=day_color_palette,\n",
    "    height=1.5,\n",
    "    aspect=0.3,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"MYH7 expression\",\n",
    "    bins=bins,\n",
    "    kde=False,\n",
    "    norm_hist=True,\n",
    "\n",
    ")\n",
    "g.set(xlabel='', ylabel='Probability density')\n",
    "g.set(xlim=(-0.3, 0.3));\n",
    "g.set(ylim=(0, None));\n",
    "g.set(xticklabels=[]);\n",
    "\n",
    "# save png and svg\n",
    "plt.savefig(save_dir_pngs/'fig_2c_rnaseq_MYH7_hist_zeros.png', dpi=100, bbox_inches = \"tight\")\n",
    "plt.savefig(save_dir_svgs/'fig_2c_rnaseq_MYH7_hist_zeros.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3c score histogram"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(4, 3))\n",
    "\n",
    "plot_3c = sns.countplot(\n",
    "    data=df_org_scores_all,\n",
    "    x=\"Expert structural annotation score (all)\",\n",
    "    hue=\"Cell age\",\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    ")\n",
    "\n",
    "plot_3c.set(ylabel='Count');\n",
    "\n",
    "sns.despine()\n",
    "plt.legend(bbox_to_anchor=(1.4, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_3c_expert_annotations_hist.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_3c_expert_annotations_hist.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3d median intensity by annotation class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.catplot(\n",
    "    data=df_org_scores_all,\n",
    "    x='Expert structural annotation score (all)',\n",
    "    y='Alpha-actinin intensity (median, normalized per day)',\n",
    "    hue='Expert structural annotation score (all)',\n",
    "    hue_order=[1,2,3,4,5],\n",
    "    s=1,\n",
    "    col=\"Cell age\",\n",
    "    kind=\"swarm\",\n",
    "    height=4,\n",
    "    aspect=1.0,\n",
    "    zorder=0\n",
    ");\n",
    "\n",
    "g.set_titles(\"Cell age = {col_name}\")  # use this argument literally\n",
    "\n",
    "axes=g.axes[0]\n",
    "\n",
    "for i, cell_age in enumerate(sorted(df[\"Cell age\"].unique())):\n",
    "\n",
    "    h = sns.boxplot(\n",
    "        data=df_org_scores_all[df_org_scores_all[\"Cell age\"]==cell_age],\n",
    "        x='Expert structural annotation score (all)',\n",
    "        y='Alpha-actinin intensity (median, normalized per day)',\n",
    "        showcaps=True,\n",
    "        boxprops={'facecolor':'None'},\n",
    "        showfliers=False,\n",
    "        ax=axes[i]\n",
    "    )\n",
    "    \n",
    "    h.set(\n",
    "        xlabel='Expert annotation score',\n",
    "        ylabel='Alpha-actinin intensity (median)'\n",
    "    )\n",
    "    \n",
    "    if i > 0:\n",
    "        h.set(\n",
    "        ylabel=''\n",
    "    )\n",
    "\n",
    "axes = g.axes[0];\n",
    "axes[0].set_ylim(0, 4);\n",
    "axes[1].set_ylim(0, 4);\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_3d_intensity_vs_expert_score_swarmbox_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_3d_intensity_vs_expert_score_swarmbox_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3 supp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import confusion_matrix\n",
    "\n",
    "conf_mat = confusion_matrix(df['Expert structural annotation score (annotator 2)'], df['Expert structural annotation score (annotator 1)'])\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "conf_mat_plot = sns.heatmap(\n",
    "    conf_mat,\n",
    "    annot=True,\n",
    "    fmt=\"d\",\n",
    "    square=True,\n",
    "    cmap=\"Blues\",\n",
    "    vmin=0,\n",
    "    vmax=1000,\n",
    "    cbar=False,\n",
    ")\n",
    "\n",
    "conf_mat_plot.set(\n",
    "    xlabel='Expert annotator 1',\n",
    "    ylabel='Expert annotator 2'\n",
    ");\n",
    "conf_mat_plot.set_xticklabels([1,2,3,4,5])\n",
    "conf_mat_plot.set_yticklabels([1,2,3,4,5]);\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_3supp_expert_annotations_confmat.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_3supp_expert_annotations_confmat.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3g stripes vs global alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(4, 3))\n",
    "\n",
    "plot_org_vs_stripes = sns.scatterplot(\n",
    "    data=df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Fraction cell area organized z-disks\",\n",
    "    y=\"Max coefficient var\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    linewidth=0,\n",
    "    alpha = 0.5,\n",
    "    s=10,\n",
    ")\n",
    "plot_org_vs_stripes.set(ylabel=\"Global alignment measure\");\n",
    "plot_org_vs_stripes.set(xlim=(0, 1));\n",
    "plot_org_vs_stripes.set(ylim=(0, 1));\n",
    "\n",
    "handles, _ = plot_org_vs_stripes.get_legend_handles_labels()\n",
    "plot_org_vs_stripes.legend(handles, [\"Cell age\", \"18\", \"32\"], bbox_to_anchor=(1.0, 0.7), frameon=False)\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_3g_regular_stripes_vs_global_alignment.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_3g_regular_stripes_vs_global_alignment.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4b regression coefs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_boot_reg = boot_regression(\n",
    "    df,\n",
    "    N=1000,\n",
    "    X_cols=BAR_PLOT_COLUMNS,\n",
    "    y_col=\"Expert structural annotation score (mean)\",\n",
    "    weight_col=\"Expert structural annotation score (mean)\",\n",
    ")\n",
    "\n",
    "reg_plot_ci_df = make_reg_plot_ci_df(df_boot_reg)\n",
    "\n",
    "fig_bar, ax_bar = make_regression_bar_plot(reg_plot_ci_df, title=\"\")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4b_linear_model_feature_importances.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4b_linear_model_feature_importances.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4c expert score vs linear model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "g = sns.stripplot(\n",
    "    data=df_org_scores_all.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Expert structural annotation score (all)\",\n",
    "    y=\"Combined organizational score\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    alpha=0.2,\n",
    "    linewidth=0,\n",
    "    jitter=0.25,\n",
    "    s=3\n",
    ")\n",
    "\n",
    "sns.despine()\n",
    "plt.legend(bbox_to_anchor=(1.4, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4c_linear_model_pred_vs_true.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4c_linear_model_pred_vs_true.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4 supp confusion matrices for linear model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conf_mat_all = confusion_matrix(df_org_scores_all['Expert structural annotation score (all)'], df_org_scores_all['Combined organizational score (rounded)'])\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "conf_mat_plot_all = sns.heatmap(\n",
    "    conf_mat_all,\n",
    "    annot=True,\n",
    "    fmt=\"d\",\n",
    "    square=True,\n",
    "    cmap=\"Blues\",\n",
    "    vmin=0,\n",
    "    vmax=2000,\n",
    "    cbar=False,\n",
    ")\n",
    "\n",
    "conf_mat_plot_all.set(\n",
    "    xlabel='Combined organizational score (rounded)',\n",
    "    ylabel='Expert structural annotation score (all)'\n",
    ");\n",
    "conf_mat_plot_all.set_xticklabels([1,2,3,4,5])\n",
    "conf_mat_plot_all.set_yticklabels([1,2,3,4,5]);\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4supp_linear_model_conf_mat_vs_both.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4supp_linear_model_conf_mat_vs_both.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conf_mat_ann_1 = confusion_matrix(df['Expert structural annotation score (annotator 1)'], df['Combined organizational score (rounded)'])\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "conf_mat_plot_ann_1 = sns.heatmap(\n",
    "    conf_mat_ann_1,\n",
    "    annot=True,\n",
    "    fmt=\"d\",\n",
    "    square=True,\n",
    "    cmap=\"Blues\",\n",
    "    vmin=0,\n",
    "    vmax=1000,\n",
    "    cbar=False,\n",
    ")\n",
    "\n",
    "conf_mat_plot_ann_1.set(\n",
    "    xlabel='Combined organizational score (rounded)',\n",
    "    ylabel='Expert structural annotation score (annotator 1)'\n",
    ");\n",
    "conf_mat_plot_ann_1.set_xticklabels([1,2,3,4,5])\n",
    "conf_mat_plot_ann_1.set_yticklabels([1,2,3,4,5]);\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4supp_linear_model_conf_mat_vs_ann_1.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4supp_linear_model_conf_mat_vs_ann_1.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conf_mat_ann_2 = confusion_matrix(df['Expert structural annotation score (annotator 2)'], df['Combined organizational score (rounded)'])\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "conf_mat_plot_ann_2 = sns.heatmap(\n",
    "    conf_mat_ann_2,\n",
    "    annot=True,\n",
    "    fmt=\"d\",\n",
    "    square=True,\n",
    "    cmap=\"Blues\",\n",
    "    vmin=0,\n",
    "    vmax=1000,\n",
    "    cbar=False,\n",
    ")\n",
    "\n",
    "conf_mat_plot_ann_2.set(\n",
    "    xlabel='Combined organizational score (rounded)',\n",
    "    ylabel='Expert structural annotation score (annotator 2)'\n",
    ");\n",
    "conf_mat_plot_ann_2.set_xticklabels([1,2,3,4,5])\n",
    "conf_mat_plot_ann_2.set_yticklabels([1,2,3,4,5]);\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4supp_linear_model_conf_mat_vs_ann_2.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4supp_linear_model_conf_mat_vs_ann_2.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,31)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    df,\n",
    "    hue='Expert structural annotation score (grouped)',\n",
    "    hue_order=[\"1-2\", \"3\", \"4-5\"],\n",
    "    height=4\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    'Combined organizational score',\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(xlim=(0, 6));\n",
    "g._legend.set(bbox_to_anchor=(0.75, 0.6))\n",
    "g._legend.set_title(\"Expert score  \")\n",
    "g.set(ylabel='Probability density');\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4e_linear_model_distplot_grouped_by_human_score.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4e_linear_model_distplot_grouped_by_human_score.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,1,31)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    df,\n",
    "    hue=\"Expert structural annotation score (grouped)\",\n",
    "    hue_order=[\"1-2\", \"3\", \"4-5\"],\n",
    "    height=4\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    'Fraction cell area organized z-disks',\n",
    "    bins=bins,\n",
    ").add_legend()\n",
    "\n",
    "g.set(xlim=(0, 1));\n",
    "g.set(ylim=(0, 25));\n",
    "g._legend.set(bbox_to_anchor=(0.75, 0.6))\n",
    "g._legend.set_title(\"Expert score      \")\n",
    "\n",
    "g.set(ylabel='Probability density');\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4e_regular_stripes_distplot_grouped_by_human_score.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4e_regular_stripes_distplot_grouped_by_human_score.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,31)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    df,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=4\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    'Combined organizational score',\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(\n",
    "    xlim=(0, 6),\n",
    "    ylabel='Probability density'\n",
    ");\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4f_linear_model_distplot_grouped_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4f_linear_model_distplot_grouped_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,1,31)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    df,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=4\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"Fraction cell area organized z-disks\",\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(\n",
    "    xlim=(0, 1),\n",
    "    ylabel='Probability density'\n",
    ");\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4f_regular_stripes_distplot_grouped_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4f_regular_stripes_distplot_grouped_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4 supp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lim_map={\n",
    "    'Cell area (μm^2)': (0.0, 10_000),\n",
    "     'Cell aspect ratio': (0.0, 1.0),\n",
    "     'Fraction cell area background': (0.0, 1.0),\n",
    "     'Fraction cell area diffuse/other': (0.0, 1.0),\n",
    "     'Fraction cell area fibers': (0.0, 1.0),\n",
    "     'Fraction cell area disorganized puncta': (0.0, 1.0),\n",
    "     'Fraction cell area organized puncta': (0.0, 1.0),\n",
    "     'Fraction cell area organized z-disks': (0.0, 1.0),\n",
    "     'Max coefficient var': (0.0, 1.0),\n",
    "     'Peak height': (0.0, 1.0),\n",
    "     'Peak distance (μm)': (0.5, 3.0),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for feature in BAR_PLOT_COLUMNS:\n",
    "\n",
    "    bins = np.linspace(0,lim_map[feature][1],21)\n",
    "    \n",
    "    g = sns.FacetGrid(\n",
    "        df,\n",
    "        hue=\"Cell age\",\n",
    "        hue_order=[18, 32],\n",
    "        palette=DAY_COLOR_PALETTE,\n",
    "        height=4\n",
    "    );\n",
    "    g = g.map(\n",
    "        sns.distplot,\n",
    "        feature,\n",
    "        bins=bins,\n",
    "        kde_kws=dict(bw=bins[-1]/len(bins))\n",
    "    ).add_legend();\n",
    "    \n",
    "    row = reg_plot_ci_df[reg_plot_ci_df[\"Feature\"]==feature].iloc[0,:]\n",
    "    title = f'Organizational score feature weight = {row[\"Feature weight (mean)\"]:.2f}\\n95% CI = ({row[\"Feature weight (CI low)\"]:.2f}, {row[\"Feature weight (CI high)\"]:.2f})'\n",
    "    \n",
    "    g.set(\n",
    "        xlim=lim_map[feature],\n",
    "        xlabel=feature,\n",
    "        ylabel='Probability density',\n",
    "        title=title,\n",
    "    );\n",
    "\n",
    "    g._legend.set_title(\"Cell age\");\n",
    "    \n",
    "    # save png and svg\n",
    "    if SAVE:\n",
    "        plt.savefig(fig4_supp_dir_pngs/f'fig_4supp_{safe(feature)}_distplot_grouped_by_day.png', dpi=100, bbox_inches = \"tight\");\n",
    "        plt.savefig(fig4_supp_dir_svgs/f'fig4_supp_{safe(feature)}_distplot_grouped_by_day.svg', format=\"svg\", bbox_inches = \"tight\");"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for feature in BAR_PLOT_COLUMNS:\n",
    "\n",
    "    bins = np.linspace(0,lim_map[feature][1],21)\n",
    "\n",
    "    g = sns.FacetGrid(\n",
    "        df,\n",
    "        hue='Expert structural annotation score (grouped)',\n",
    "        hue_order=[\"1-2\", \"3\", \"4-5\"],\n",
    "        height=4\n",
    "    );\n",
    "    g = g.map(\n",
    "        sns.distplot,\n",
    "        feature,\n",
    "        bins=bins,\n",
    "        kde_kws=dict(bw=bins[-1]/len(bins))\n",
    "    ).add_legend();\n",
    "\n",
    "    row = reg_plot_ci_df[reg_plot_ci_df[\"Feature\"]==feature].iloc[0,:]\n",
    "    title = f'Organizational score feature weight = {row[\"Feature weight (mean)\"]:.2f}\\n95% CI = ({row[\"Feature weight (CI low)\"]:.2f}, {row[\"Feature weight (CI high)\"]:.2f})'\n",
    "    \n",
    "    g.set(\n",
    "        xlim=lim_map[feature],\n",
    "        xlabel=feature.replace(\"_\", \" \").capitalize(),\n",
    "        ylabel='Probability density',\n",
    "        title=title\n",
    "    );\n",
    "    g._legend.set(bbox_to_anchor=(0.6, 0.6))\n",
    "    g._legend.set_title(\"Expert score    \")\n",
    "\n",
    "    # save png and svg\n",
    "    if SAVE:\n",
    "        plt.savefig(fig4_supp_dir_pngs/f'fig_4supp_{safe(feature)}_distplot_grouped_by_human_score.png', dpi=100, bbox_inches = \"tight\");\n",
    "        plt.savefig(fig4_supp_dir_svgs/f'fig_4supp_{safe(feature)}_distplot_grouped_by_human_score.svg', format=\"svg\", bbox_inches = \"tight\");"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4 g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.FacetGrid(\n",
    "    df,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=4,\n",
    "    aspect=1,\n",
    "    col=\"Cell age\",\n",
    "    sharey=False,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.scatterplot,\n",
    "    \"Combined organizational score\",\n",
    "    \"Alpha-actinin intensity (median, normalized per day)\",\n",
    "    linewidth=0,\n",
    "    alpha = 0.5,\n",
    "    s=10,\n",
    ").add_legend(title='Cell age')\n",
    "\n",
    "axes = g.axes[0];\n",
    "axes[0].set_ylim(0, 3);\n",
    "axes[1].set_ylim(0, 3);\n",
    "\n",
    "g.set_titles(\"Cell age = {col_name}\")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4g_linear_model_org_vs_actn2_medain_intensity.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4g_linear_model_org_vs_actn2_medain_intensity.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x='Combined organizational score'\n",
    "y='Alpha-actinin intensity (median, normalized per day)'\n",
    "stratify='Cell age'\n",
    "\n",
    "df_int_org_corr = boot_spearmanr(df, x=x, y=y, stratify=stratify, N=1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (1, 3)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "bar_order = [\"18\", \"32\"]\n",
    "\n",
    "feat_chart = sns.barplot(\n",
    "    data=df_int_org_corr,\n",
    "    x=stratify,\n",
    "    order=bar_order,\n",
    "    y=f\"Spearman R ({x}, {y}) (mean)\",\n",
    "    hue=stratify,\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    dodge=False\n",
    ");\n",
    "\n",
    "mean_col = f\"Spearman R ({x}, {y}) (mean)\"\n",
    "hi_col = f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "low_col = f\"Spearman R ({x}, {y}) (CI low)\"\n",
    "\n",
    "err = np.array(((df_int_org_corr[mean_col] - df_int_org_corr[low_col]).values,\n",
    "                (df_int_org_corr[hi_col] - df_int_org_corr[mean_col]).values))\n",
    "\n",
    "plt.errorbar(\n",
    "    x=df_int_org_corr.set_index(stratify).loc[bar_order,:].reset_index()[stratify],\n",
    "    y=df_int_org_corr.set_index(stratify).loc[bar_order,:].reset_index()[mean_col],\n",
    "    fmt='none',\n",
    "    yerr=err,\n",
    "    ecolor='k',\n",
    "    elinewidth=1,\n",
    "    capsize=4\n",
    ")\n",
    "\n",
    "sns.despine();\n",
    "\n",
    "# feat_chart.set_xticklabels(feat_chart.get_xticklabels(), rotation=45, horizontalalignment='right');\n",
    "feat_chart.set(\n",
    "    xlabel=stratify,\n",
    "    ylabel=f\"Spearman R:\\n{x}\\n{y}\"\n",
    ");\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.7), frameon=False);\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_4g_linear_model_org_vs_actn2_median_intensity_corr.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_4g_linear_model_org_vs_actn2_median_intensity_corr.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5a probe density vs structure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.FacetGrid(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    col=\"FISH probe\",\n",
    "    col_order=PROBE_ORDER,\n",
    "    col_wrap=4,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    sharex=True,\n",
    "    xlim=(1,5),\n",
    "    sharey=False,\n",
    "    height=3.0,\n",
    "    aspect=1.0,\n",
    ")\n",
    "\n",
    "g = g.map(\n",
    "    plt.scatter,\n",
    "    \"Combined organizational score\",\n",
    "    \"FISH probe (count/μm^2)\",\n",
    "    s=10,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    ").add_legend()\n",
    "\n",
    "g.set_titles(\"{col_name}\")\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5a_probe_density_vs_org_score.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5a_probe_density_vs_org_score.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.FacetGrid(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    col=\"FISH probe\",\n",
    "    col_order=PROBE_ORDER,\n",
    "    col_wrap=4,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    sharex=True,\n",
    "    xlim=(1,5),\n",
    "    sharey=False,\n",
    "    height=3.0,\n",
    "    aspect=1.0,\n",
    ")\n",
    "\n",
    "g = g.map(\n",
    "    plt.scatter,\n",
    "    \"Combined organizational score\",\n",
    "    \"FISH probe count\",\n",
    "    s=10,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    ").add_legend()\n",
    "\n",
    "g.set_titles(\"{col_name}\")\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_supp_probe_count_vs_org_score.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_supp_probe_count_vs_org_score.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5b spearman correlation bar charts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_density_time_stat = pd.DataFrame()\n",
    "\n",
    "for probe in df_tidy[\"FISH probe\"].unique():\n",
    "    df_tmp = df_tidy[df_tidy[\"FISH probe\"] == probe][[\"FISH probe (count/μm^2)\", \"Cell age\"]]\n",
    "    early_densities = df_tmp[df_tmp[\"Cell age\"] == 18][\"FISH probe (count/μm^2)\"]\n",
    "    late_densities = df_tmp[df_tmp[\"Cell age\"] == 32][\"FISH probe (count/μm^2)\"]\n",
    "    df_tmp_stat = pg.mwu(early_densities, late_densities).reset_index().rename(columns={\"index\":\"Statistical test\"})\n",
    "    df_tmp_stat[\"FISH probe\"] = probe\n",
    "    df_density_time_stat = df_density_time_stat.append(df_tmp_stat)\n",
    "    \n",
    "df_density_time_stat = df_density_time_stat.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "N_boot=1000\n",
    "\n",
    "x = \"Combined organizational score\"\n",
    "y = \"FISH probe (count/μm^2)\"\n",
    "col = \"FISH probe\"\n",
    "hue = \"Cell age\"\n",
    "\n",
    "\n",
    "df_corr_bar_in = df_tidy[[col, hue, x, y]].copy()\n",
    "df_corr_bar = pd.DataFrame(columns = [col, hue, f\"Spearman R ({x}, {y})\", \"Spearman p-value\"])\n",
    "\n",
    "N_boot=1000\n",
    "\n",
    "for i in range(N_boot):\n",
    "    df_corr_bar_boot = df_corr_bar_in.sample(frac=1, replace=True).reset_index(drop=True)\n",
    "    for probe in df_tidy[col].unique():\n",
    "\n",
    "        condition = (df_corr_bar_boot[col] == probe)\n",
    "        df_tmp = df_corr_bar_boot[condition]\n",
    "        rho = spearmanr(df_tmp[x], df_tmp[y])\n",
    "        df_corr_bar = df_corr_bar.append(\n",
    "            pd.DataFrame({\n",
    "                \"FISH probe\": [probe],\n",
    "                \"Cell age\": [\"all\"],\n",
    "                f\"Spearman R ({x}, {y})\": [rho[0]],\n",
    "                \"Spearman p-value\": [rho[1]],\n",
    "            })\n",
    "        )\n",
    "\n",
    "        for age in df_tidy[hue].unique():\n",
    "            condition = (df_corr_bar_boot[col] == probe) & (df_corr_bar_boot[hue] == age)\n",
    "            df_tmp = df_corr_bar_boot[condition]\n",
    "            rho = spearmanr(df_tmp[x], df_tmp[y])\n",
    "            df_corr_bar = df_corr_bar.append(\n",
    "                pd.DataFrame({\n",
    "                    \"FISH probe\": [probe],\n",
    "                    \"Cell age\": [age],\n",
    "                    f\"Spearman R ({x}, {y})\": [rho[0]],\n",
    "                    \"Spearman p-value\": [rho[1]],\n",
    "                })\n",
    "            )\n",
    "\n",
    "df_corr_bar = df_corr_bar.reset_index(drop=True)\n",
    "\n",
    "f = {f\"Spearman R ({x}, {y})\": ['mean', ci_low, ci_high]}\n",
    "df_corr_bar_plot = df_corr_bar[[col, hue, f\"Spearman R ({x}, {y})\"]].groupby([col, hue]).agg(f).reset_index()\n",
    "df_corr_bar_plot.columns = [\" \".join(x).strip() for x in list(zip(*[list(df_corr_bar_plot.columns.get_level_values(i)) for i in (0,1)]))]\n",
    "\n",
    "df_corr_bar_plot = df_corr_bar_plot.rename(\n",
    "    columns={\n",
    "        f\"Spearman R ({x}, {y}) mean\": f\"Spearman R ({x}, {y}) (mean)\",\n",
    "        f\"Spearman R ({x}, {y}) ci_low\": f\"Spearman R ({x}, {y}) (CI low)\",\n",
    "        f\"Spearman R ({x}, {y}) ci_high\": f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(8, 4))\n",
    "\n",
    "df_corr_bar_plot[hue] = df_corr_bar_plot[hue].astype(str)\n",
    "\n",
    "df_corr_bar_plot_all = df_corr_bar_plot[df_corr_bar_plot[hue] == \"all\"].set_index(col).loc[PROBE_ORDER, :].reset_index()\n",
    "df_corr_bar_plot_18 = df_corr_bar_plot[df_corr_bar_plot[hue] == \"18\"].set_index(col).loc[PROBE_ORDER, :].reset_index()\n",
    "df_corr_bar_plot_32 = df_corr_bar_plot[df_corr_bar_plot[hue] == \"32\"].set_index(col).loc[PROBE_ORDER, :].reset_index()\n",
    "\n",
    "# set width of bar\n",
    "barWidth = 0.25\n",
    " \n",
    "mean_col = f\"Spearman R ({x}, {y}) (mean)\"\n",
    "hi_col = f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "low_col = f\"Spearman R ({x}, {y}) (CI low)\"\n",
    "\n",
    "# set height of bar\n",
    "bars_all = df_corr_bar_plot_all[mean_col].values\n",
    "bars_18 = df_corr_bar_plot_18[mean_col].values\n",
    "bars_32 = df_corr_bar_plot_32[mean_col].values\n",
    "\n",
    "# set error bars for each series\n",
    "err_all = np.array(((df_corr_bar_plot_all[mean_col] - df_corr_bar_plot_all[low_col]).values,\n",
    "                    (df_corr_bar_plot_all[hi_col] - df_corr_bar_plot_all[mean_col]).values))\n",
    "\n",
    "err_18 = np.array(((df_corr_bar_plot_18[mean_col] - df_corr_bar_plot_18[low_col]).values,\n",
    "                   (df_corr_bar_plot_18[hi_col] - df_corr_bar_plot_18[mean_col]).values))\n",
    "\n",
    "err_32 = np.array(((df_corr_bar_plot_32[mean_col] - df_corr_bar_plot_32[low_col]).values,\n",
    "                   (df_corr_bar_plot_32[hi_col] - df_corr_bar_plot_32[mean_col]).values))\n",
    "\n",
    " \n",
    "# Set position of bar on X axis\n",
    "r_all = np.arange(len(bars_all))\n",
    "r_18 = [x + barWidth for x in r_all]\n",
    "r_32 = [x + barWidth for x in r_18]\n",
    " \n",
    "# Make the plot\n",
    "plt.bar(r_all, bars_all, yerr=err_all, color='grey',    width=barWidth, capsize=4, edgecolor='white', label='all', error_kw=dict(elinewidth=1))\n",
    "plt.bar(r_18, bars_18, yerr=err_18, color=DAY_18_COLOR, width=barWidth, capsize=4, edgecolor='white', label='18',  error_kw=dict(elinewidth=1))\n",
    "plt.bar(r_32, bars_32, yerr=err_32, color=DAY_32_COLOR, width=barWidth, capsize=4, edgecolor='white', label='32',  error_kw=dict(elinewidth=1))\n",
    " \n",
    "# Add xticks on the middle of the group bars\n",
    "plt.xlabel(col)\n",
    "plt.xticks([r + barWidth for r in range(len(bars_all))], df_corr_bar_plot_all[col].unique())\n",
    "plt.ylim((-0.6, 0.6))\n",
    "plt.ylabel(f\"Spearman R:\\n{x},\\n{y}\")\n",
    "\n",
    "# Create legend & Show graphic\n",
    "plt.legend(bbox_to_anchor=(1.2, 0.6),frameon=False)\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5b_gnxp_vs_struct_corr.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5b_gnxp_vs_struct_corr.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "N_boot=1000\n",
    "\n",
    "x = \"Combined organizational score\"\n",
    "y = \"FISH probe count\"\n",
    "col = \"FISH probe\"\n",
    "hue = \"Cell age\"\n",
    "\n",
    "df_corr_bar_in = df_tidy[[col, hue, x, y]].copy()\n",
    "df_corr_bar = pd.DataFrame(columns = [col, hue, f\"Spearman R ({x}, {y})\", \"Spearman R p-value\"])\n",
    "\n",
    "for i in range(N_boot):\n",
    "    df_corr_bar_boot = df_corr_bar_in.sample(frac=1, replace=True).reset_index(drop=True)\n",
    "    for probe in df_tidy[col].unique():\n",
    "\n",
    "        condition = (df_corr_bar_boot[col] == probe)\n",
    "        df_tmp = df_corr_bar_boot[condition]\n",
    "        rho = spearmanr(df_tmp[x], df_tmp[y])\n",
    "        df_corr_bar = df_corr_bar.append(\n",
    "            pd.DataFrame({\n",
    "                f\"{col}\": [probe],\n",
    "                f\"{hue}\": [\"all\"],\n",
    "                f\"Spearman R ({x}, {y})\": [rho[0]],\n",
    "                \"Spearman p-value\": [rho[1]],\n",
    "            })\n",
    "        )\n",
    "\n",
    "        for age in df_tidy[hue].unique():\n",
    "            condition = (df_corr_bar_boot[col] == probe) & (df_corr_bar_boot[hue] == age)\n",
    "            df_tmp = df_corr_bar_boot[condition]\n",
    "            rho = spearmanr(df_tmp[x], df_tmp[y])\n",
    "            df_corr_bar = df_corr_bar.append(\n",
    "                pd.DataFrame({\n",
    "                    f\"{col}\": [probe],\n",
    "                    f\"{hue}\": [age],\n",
    "                    f\"Spearman R ({x}, {y})\": [rho[0]],\n",
    "                    \"Spearman p-value\": [rho[1]],\n",
    "                })\n",
    "            )\n",
    "\n",
    "df_corr_bar = df_corr_bar.reset_index(drop=True)\n",
    "\n",
    "f = {f\"Spearman R ({x}, {y})\": ['mean', ci_low, ci_high]}\n",
    "df_corr_bar_plot = df_corr_bar[[f\"{col}\", f\"{hue}\", f\"Spearman R ({x}, {y})\"]].groupby([col, hue]).agg(f).reset_index()\n",
    "df_corr_bar_plot.columns = [\" \".join(x).strip() for x in list(zip(*[list(df_corr_bar_plot.columns.get_level_values(i)) for i in (0,1)]))]\n",
    "\n",
    "df_corr_bar_plot = df_corr_bar_plot.rename(\n",
    "    columns={\n",
    "        f\"Spearman R ({x}, {y}) mean\": f\"Spearman R ({x}, {y}) (mean)\",\n",
    "        f\"Spearman R ({x}, {y}) ci_low\": f\"Spearman R ({x}, {y}) (CI low)\",\n",
    "        f\"Spearman R ({x}, {y}) ci_high\": f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(8, 4))\n",
    "\n",
    "df_corr_bar_plot[hue] = df_corr_bar_plot[hue].astype(str)\n",
    "\n",
    "df_corr_bar_plot_all = df_corr_bar_plot[df_corr_bar_plot[hue] == \"all\"].set_index(\"FISH probe\").loc[PROBE_ORDER, :].reset_index()\n",
    "df_corr_bar_plot_18 = df_corr_bar_plot[df_corr_bar_plot[hue] == \"18\"].set_index(\"FISH probe\").loc[PROBE_ORDER, :].reset_index()\n",
    "df_corr_bar_plot_32 = df_corr_bar_plot[df_corr_bar_plot[hue] == \"32\"].set_index(\"FISH probe\").loc[PROBE_ORDER, :].reset_index()\n",
    "\n",
    "# set width of bar\n",
    "barWidth = 0.25\n",
    "\n",
    "mean_col = f\"Spearman R ({x}, {y}) (mean)\"\n",
    "hi_col = f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "low_col = f\"Spearman R ({x}, {y}) (CI low)\"\n",
    "\n",
    "# set height of bar\n",
    "bars_all = df_corr_bar_plot_all[mean_col].values\n",
    "bars_18 = df_corr_bar_plot_18[mean_col].values\n",
    "bars_32 = df_corr_bar_plot_32[mean_col].values\n",
    "\n",
    "# set error bars for each series\n",
    "err_all = np.array(((df_corr_bar_plot_all[mean_col] - df_corr_bar_plot_all[low_col]).values,\n",
    "                    (df_corr_bar_plot_all[hi_col] - df_corr_bar_plot_all[mean_col]).values))\n",
    "\n",
    "err_18 = np.array(((df_corr_bar_plot_18[mean_col] - df_corr_bar_plot_18[low_col]).values,\n",
    "                   (df_corr_bar_plot_18[hi_col] - df_corr_bar_plot_18[mean_col]).values))\n",
    "\n",
    "err_32 = np.array(((df_corr_bar_plot_32[mean_col] - df_corr_bar_plot_32[low_col]).values,\n",
    "                   (df_corr_bar_plot_32[hi_col] - df_corr_bar_plot_32[mean_col]).values))\n",
    "\n",
    "# Set position of bar on X axis\n",
    "r_all = np.arange(len(bars_all))\n",
    "r_18 = [x + barWidth for x in r_all]\n",
    "r_32 = [x + barWidth for x in r_18]\n",
    "\n",
    "# Make the plot\n",
    "plt.bar(r_all, bars_all, yerr=err_all, color='grey',       width=barWidth, capsize=4, edgecolor='white', label='all', error_kw=dict(elinewidth=1))\n",
    "plt.bar(r_18, bars_18, yerr=err_18, color=DAY_18_COLOR, width=barWidth, capsize=4, edgecolor='white', label='18', error_kw=dict(elinewidth=1))\n",
    "plt.bar(r_32, bars_32, yerr=err_32, color=DAY_32_COLOR, width=barWidth, capsize=4, edgecolor='white', label='32', error_kw=dict(elinewidth=1))\n",
    "\n",
    "# Add xticks on the middle of the group bars\n",
    "plt.xlabel(col)\n",
    "plt.xticks([r + barWidth for r in range(len(bars_all))], df_corr_bar_plot_all[\"FISH probe\"].unique())\n",
    "plt.ylim((-0.6, 0.6))\n",
    "plt.ylabel(f\"Spearman R:\\n{x},\\n{y}\")\n",
    "\n",
    "# Create legend & Show graphic\n",
    "plt.legend(bbox_to_anchor=(1.2, 0.6),frameon=False)\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_supp_probe_count_vs_struct_corr.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_supp_probe_count_vs_struct_corr.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5c myh7-myh6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = \"Combined organizational score\"\n",
    "y = \"MYH7-MYH6 relative transcript abundance (normalized)\"\n",
    "stratify = \"Cell age\"\n",
    "\n",
    "plt.figure(figsize=(4, 4))\n",
    "\n",
    "plot_myh67_vs_org_score = sns.scatterplot(\n",
    "    data=df.dropna(subset=[x,y]),\n",
    "    x=x,\n",
    "    y=y,\n",
    "    hue=stratify,\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    linewidth=0,\n",
    "    alpha = 0.5,\n",
    "    s=15\n",
    ")\n",
    "plot_myh67_vs_org_score.set(\n",
    "    xlim=(1, 5),\n",
    "    ylim=(-1, 1),\n",
    "    ylabel=y.replace(\"transcript abundance\", \"transcript\\nabundance\")\n",
    ");\n",
    "\n",
    "handles, _ = plot_myh67_vs_org_score.get_legend_handles_labels()\n",
    "plot_myh67_vs_org_score.legend(handles, [\"Cell age\", \"18\", \"32\"], bbox_to_anchor=(1.0, 0.7), frameon=False)\n",
    "\n",
    "\n",
    "df_boot_spr = boot_spearmanr(df, x=x, y=y, stratify=stratify, N=1000)\n",
    "\n",
    "mean_col = f\"Spearman R ({x}, {y}) (mean)\"\n",
    "hi_col = f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "low_col = f\"Spearman R ({x}, {y}) (CI low)\"\n",
    "\n",
    "title = \"\\n\".join([\n",
    "    f\"Spearman R day {day} = {row[mean_col]:.2f}, 95% CI = ({row[low_col]:.2f}, {row[hi_col]:.2f})\"\n",
    "    for day, row in df_boot_spr.set_index(stratify).iterrows()\n",
    "])\n",
    "\n",
    "plt.title(title)\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5c_linear_model_vs_myh_diff_scatter.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5c_linear_model_vs_myh_diff_scatter.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dummy = pd.concat([df,pd.DataFrame(np.zeros(len(df)), columns=[\"MYH67 vs org score all cells\"])], axis=1)\n",
    "boot_spearmanr(df_dummy, x=x, y=y, stratify=\"MYH67 vs org score all cells\", N=1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(1,5,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=1.5,\n",
    "    aspect=2.75,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"Combined organizational score\",\n",
    "    bins=bins,\n",
    "    kde_kws=dict(bw=0.1)\n",
    ")\n",
    "g.set(ylabel='Probability density')\n",
    "g.set(xlim=(1, 5));\n",
    "g.set(ylim=(0, 0.7));\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5c_linear_model_vs_myh_diff_top_hist.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5c_linear_model_vs_myh_diff_top_hist.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(-1,1,51)\n",
    "bins -= (bins[1]-bins[0])/2\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=1.5,\n",
    "    aspect=2.75,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"MYH7-MYH6 relative transcript abundance (normalized)\",\n",
    "    bins=bins,\n",
    "    kde_kws=dict(bw=0.1)\n",
    ")\n",
    "g.set(ylabel='Probability density')\n",
    "g.set(xlim=(-1,1));\n",
    "g.set(ylim=(0, 1.5));\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5c_linear_model_vs_myh_diff_right_hist.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5c_linear_model_vs_myh_diff_right_hist.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5c supplement -- myh67 diff from feats directly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_boot_reg = boot_regression(\n",
    "    df.dropna(subset=[\"MYH7-MYH6 relative transcript abundance (normalized)\"]),\n",
    "    N=1000,\n",
    "    X_cols=BAR_PLOT_COLUMNS,\n",
    "    y_col=\"MYH7-MYH6 relative transcript abundance (normalized)\",\n",
    "    weight_col=None,\n",
    ")\n",
    "\n",
    "reg_plot_ci_df = make_reg_plot_ci_df(df_boot_reg)\n",
    "\n",
    "fig_bar, ax_bar = make_regression_bar_plot(reg_plot_ci_df, title=\"MYH7-MYH6 relative transcript abundance (normalized)\")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5f_predict_MYH67_diff_coefs.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5f_predict_MYH67_diff_coefs.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = 'MYH7-MYH6 relative transcript abundance (normalized)'\n",
    "y = 'MYH7-MYH6 relative transcript abundance (normalized) predicted'\n",
    "stratify = 'Cell age'\n",
    "\n",
    "df_pred_true  = get_pred_true(\n",
    "    df.dropna(subset=[x]).reset_index(drop=True),\n",
    "    y_col=x,\n",
    "    weight_col=None,\n",
    "    meta_cols=[stratify]\n",
    ")\n",
    "\n",
    "df_boot_spr = boot_spearmanr(df_pred_true, x=x, y=y, stratify=stratify, N=1000)\n",
    "\n",
    "mean_col = f\"Spearman R ({x}, {y}) (mean)\"\n",
    "hi_col = f\"Spearman R ({x}, {y}) (CI high)\"\n",
    "low_col = f\"Spearman R ({x}, {y}) (CI low)\"\n",
    "\n",
    "title = \"\\n\".join([\n",
    "    f\"Spearman R day {day} = {row[mean_col]:.2f}, 95% CI = ({row[low_col]:.2f}, {row[hi_col]:.2f})\"\n",
    "    for day, row in df_boot_spr.set_index(stratify).iterrows()\n",
    "])\n",
    "\n",
    "fig_scatter, ax_scatter = make_regression_scatter_plot(df_pred_true, target=x, title=title);\n",
    "ax_scatter.set(\n",
    "    xlabel=x.replace(\"MYH7-MYH6\", \"Actual MYH7-MYH6\").replace(\"relative transcript\", \"relative\\ntranscript\"),\n",
    "    ylabel=x.replace(\"MYH7-MYH6\", \"Predicted MYH7-MYH6\").replace(\"relative transcript\", \"relative\\ntranscript\")\n",
    ")\n",
    "plt.legend(bbox_to_anchor=(1.05, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5e_predict_MYH67_diff.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5e_predict_MYH67_diff.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dummy = pd.concat([df_pred_true,pd.DataFrame(np.zeros(len(df_pred_true)),\n",
    "                                                columns=[\"MYH67 pred vs true all cells\"])], axis=1)\n",
    "boot_spearmanr(df_dummy, x=x, y=y, stratify=\"MYH67 pred vs true all cells\", N=1000)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5supp count and density distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (8,3)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.violinplot(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"FISH probe\",\n",
    "    order=PROBE_ORDER,\n",
    "    y=\"FISH probe (count/μm^2)\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    split=True,\n",
    "    linewidth=1\n",
    ")\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.2, 0.7), frameon=False, title=\"Cell age\");\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5supp_probe_density_violin_plots.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5supp_probe_density_violin_plots.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (8,3)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.violinplot(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"FISH probe\",\n",
    "    order=PROBE_ORDER,\n",
    "    y=\"FISH probe count\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    split=True,\n",
    "    linewidth=1\n",
    ")\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.2, 0.7), frameon=False, title=\"Cell age\");\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_5supp_probe_count_violin_plots.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_5supp_probe_count_violin_plots.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6c & d -- hand scored sarcomeric vs not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_becky_loc_summary = pd.concat(\n",
    "    [\n",
    "        pd.concat(\n",
    "            [\n",
    "                df.dropna(\n",
    "                    subset=[f\"{x} localization\"]\n",
    "                )[f\"{x} localization\"].groupby(\n",
    "                    df.dropna(subset=[f\"{x} localization\"])[\"Cell age\"]\n",
    "                ).value_counts(\n",
    "                    normalize=True\n",
    "                ).rename(\"Fraction\").reset_index().rename(columns={f\"{x} localization\":\"Localization\"}),\n",
    "                pd.DataFrame(\n",
    "                    {\n",
    "                        \"FISH probe\": [x]*len(set(df[\"Cell age\"]))*len(set(df[\"MYH7 localization\"].dropna()) | set(df[\"MYH6 localization\"].dropna()))\n",
    "                    }\n",
    "                )\n",
    "            ], axis=\"columns\"\n",
    "        )\n",
    "        for x in [\"MYH6\", \"MYH7\"]\n",
    "]\n",
    ").reset_index(\n",
    "    drop=True\n",
    ")\n",
    "df_becky_loc_summary[\"Cell age\"] = df_becky_loc_summary[\"Cell age\"].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.dropna(subset=['MYH6 localization','MYH7 localization']).groupby(\n",
    "    [\"Cell age\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.catplot(\n",
    "    data=df_becky_loc_summary,\n",
    "    kind=\"bar\",\n",
    "    col=\"FISH probe\",\n",
    "    col_order=[\"MYH7\", \"MYH6\"],\n",
    "    x=\"Localization\",\n",
    "    order=[\"Low expression\", \"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Fraction\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[\"18\", \"32\"],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    height=3,\n",
    "    aspect=1\n",
    ");\n",
    "g.set_titles(\"{col_name}\") \n",
    "for axes in g.axes.flat:\n",
    "    axes.set_xticklabels(axes.get_xticklabels(), rotation=45, horizontalalignment='right')\n",
    "    \n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6cd_predict_MYH67_diff.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6cd_predict_MYH67_diff.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.swarmplot(\n",
    "    data=df[df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])],\n",
    "    x=\"MYH7 localization\",\n",
    "    y='Combined organizational score',\n",
    "    hue=\"MYH7 localization\",\n",
    "    hue_order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    zorder=0,\n",
    "    s=3\n",
    ")\n",
    "\n",
    "sns.boxplot(\n",
    "    data=df[df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])],\n",
    "    x=\"MYH7 localization\",\n",
    "    order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Combined organizational score\",\n",
    "    showcaps=True,\n",
    "    boxprops={'facecolor':'None'},\n",
    "    showfliers=False,\n",
    "    ax=ax\n",
    ")\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.6), frameon=False);\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6e_linear_model_org_by_becky_swarmbox.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6e_linear_model_org_by_becky_swarmbox.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sarc = df[df[\"MYH7 localization\"] == \"Sarcomeric\"][\"Combined organizational score\"]\n",
    "non_sarc = df[df[\"MYH7 localization\"] == 'Non-sarcomeric'][\"Combined organizational score\"]\n",
    "pg.mwu(sarc, non_sarc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[df[\"MYH7 localization\"].isin(\n",
    "    [\"Non-sarcomeric\", \"Sarcomeric\"]\n",
    ")].groupby(\n",
    "    [\"Cell age\"]\n",
    ").size().reset_index().rename(columns={0:'Count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_age = 18\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.swarmplot(\n",
    "    data=df[(df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])) & (df[\"Cell age\"]==cell_age)],\n",
    "    x=\"MYH7 localization\",\n",
    "    order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Combined organizational score\",\n",
    "    hue=\"MYH7 localization\",\n",
    "    hue_order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    zorder=0,\n",
    "    s=3\n",
    ")\n",
    "\n",
    "sns.boxplot(\n",
    "    data=df[(df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])) & (df[\"Cell age\"]==cell_age)],\n",
    "    x=\"MYH7 localization\",\n",
    "    order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Combined organizational score\",\n",
    "    showcaps=True,\n",
    "    boxprops={'facecolor':'None'},\n",
    "    showfliers=False,\n",
    "    ax=ax\n",
    ")\n",
    "\n",
    "ax.set(title=f\"Cell age = {cell_age}\")\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.6), frameon=False);\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_6e_linear_model_org_by_becky_swarmbox_and_day_{cell_age}.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_6e_linear_model_org_by_becky_swarmbox_and_day_{cell_age}.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_age = 18\n",
    "\n",
    "sarc = df[(df[\"MYH7 localization\"] == \"Sarcomeric\") & (df[\"Cell age\"]==cell_age)][\"Combined organizational score\"]\n",
    "non_sarc = df[(df[\"MYH7 localization\"] == \"Non-sarcomeric\") & (df[\"Cell age\"]==cell_age)][\"Combined organizational score\"]\n",
    "pg.mwu(sarc, non_sarc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_age = 32\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.swarmplot(\n",
    "    data=df[(df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])) & (df[\"Cell age\"]==cell_age)],\n",
    "    x=\"MYH7 localization\",\n",
    "    order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Combined organizational score\",\n",
    "    hue=\"MYH7 localization\",\n",
    "    hue_order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    zorder=0,\n",
    "    s=3\n",
    ")\n",
    "\n",
    "sns.boxplot(\n",
    "    data=df[(df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])) & (df[\"Cell age\"]==cell_age)],\n",
    "    x=\"MYH7 localization\",\n",
    "    order=[\"Non-sarcomeric\", \"Sarcomeric\"],\n",
    "    y=\"Combined organizational score\",\n",
    "    showcaps=True,\n",
    "    boxprops={'facecolor':'None'},\n",
    "    showfliers=False,\n",
    "    ax=ax\n",
    ")\n",
    "\n",
    "ax.set(title=f\"Cell age = {cell_age}\")\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.6), frameon=False);\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_6e_linear_model_org_by_becky_swarmbox_and_day_{cell_age}.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_6e_linear_model_org_by_becky_swarmbox_and_day_{cell_age}.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_age = 32\n",
    "\n",
    "sarc = df[(df[\"MYH7 localization\"] == \"Sarcomeric\") & (df[\"Cell age\"]==cell_age)][\"Combined organizational score\"]\n",
    "non_sarc = df[(df[\"MYH7 localization\"] == \"Non-sarcomeric\") & (df[\"Cell age\"]==cell_age)][\"Combined organizational score\"]\n",
    "pg.mwu(sarc, non_sarc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,6,21)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df[df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])],\n",
    "    hue=\"MYH7 localization\",\n",
    "    height=4,\n",
    "    aspect=1,\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    \"Combined organizational score\",\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(\n",
    "    ylabel='Probability density'\n",
    ")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6e_linear_model_org_by_becky.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6e_linear_model_org_by_becky.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6 supp c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bins = np.linspace(0,0.8,21)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df[df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])],\n",
    "    hue=\"MYH7 localization\",\n",
    "    height=4,\n",
    "    aspect=1,\n",
    "    col=\"Cell age\"\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    'MYH7 (count/μm^2)',\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(\n",
    "    ylabel='Probability density'\n",
    ")\n",
    "\n",
    "g.set_titles(\"Cell age = {col_name}\")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6supp_c_myh7_density_hist_by_sarc_and_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6supp_c_myh7_density_hist_by_sarc_and_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6 supp d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = 'Enrichment of MYH7 transcript localization to organized z-disks'\n",
    "\n",
    "bins = np.linspace(-0.2, 0.2, 11)\n",
    "\n",
    "g = sns.FacetGrid(\n",
    "    data=df[df[\"MYH7 localization\"].isin([\"Non-sarcomeric\", \"Sarcomeric\"])],\n",
    "    hue=\"MYH7 localization\",\n",
    "    height=4,\n",
    "    aspect=1,\n",
    "    col=\"Cell age\"\n",
    ")\n",
    "g = g.map(\n",
    "    sns.distplot,\n",
    "    x,\n",
    "    bins=bins\n",
    ").add_legend()\n",
    "\n",
    "g.set(\n",
    "    xlabel=x.replace(\"transcript \", \"transcript\\n\"),\n",
    "    ylabel='Probability density'\n",
    ")\n",
    "\n",
    "g.set_titles(\"Cell age = {col_name}\")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6supp_d_myh7_loc_enrichment_hist_by_sarc_and_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6supp_d_myh7_loc_enrichment_hist_by_sarc_and_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = \"Enrichment of MYH7 transcript localization to organized z-disks\"\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "ax = sns.scatterplot(\n",
    "    data=df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Combined organizational score\",\n",
    "    y=y,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    "    s=10\n",
    ")\n",
    "sns.despine();\n",
    "\n",
    "handles, _ = ax.get_legend_handles_labels()\n",
    "ax.legend(handles, [\"Cell age\", \"18\", \"32\"])\n",
    "ax.set(ylabel=y.replace(\"transcript localization\", \"transcript\\nlocalization\"))\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.05, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6f_MYH7_loc_regular_stripes_vs_linear_model_org_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6f_MYH7_loc_regular_stripes_vs_linear_model_org_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = \"Enrichment of MYH7 transcript localization to organized puncta\"\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "ax = sns.scatterplot(\n",
    "    data=df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Combined organizational score\",\n",
    "    y=y,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    "    s=10\n",
    ")\n",
    "sns.despine();\n",
    "\n",
    "handles, _ = ax.get_legend_handles_labels()\n",
    "ax.legend(handles, [\"Cell age\", \"18\", \"32\"])\n",
    "ax.set(ylabel=y.replace(\"transcript localization\", \"transcript\\nlocalization\"))\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.05, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6f_MYH7_loc_regular_dots_vs_linear_model_org_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6f_MYH7_loc_regular_dots_vs_linear_model_org_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = \"Enrichment of MYH7 transcript localization to fibers\"\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "ax = sns.scatterplot(\n",
    "    data=df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Combined organizational score\",\n",
    "    y=y,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    "    s=10\n",
    ")\n",
    "sns.despine();\n",
    "\n",
    "handles, _ = ax.get_legend_handles_labels()\n",
    "ax.legend(handles, [\"Cell age\", \"18\", \"32\"])\n",
    "ax.set(ylabel=y.replace(\"transcript localization\", \"transcript\\nlocalization\"))\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.05, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6f_MYH7_loc_threads_vs_linear_model_org_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6f_MYH7_loc_threads_vs_linear_model_org_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = \"Enrichment of MYH7 transcript localization to fibers, organized puncta, and organized z-disks\"\n",
    "\n",
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "ax = sns.scatterplot(\n",
    "    data=df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x=\"Combined organizational score\",\n",
    "    y=y,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    "    s=10\n",
    ")\n",
    "sns.despine();\n",
    "\n",
    "handles, _ = ax.get_legend_handles_labels()\n",
    "ax.legend(handles, [\"Cell age\", \"18\", \"32\"])\n",
    "ax.set(ylabel=y.replace(\"to \", \"to\\n\"))\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.05, 0.6),frameon=False)\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_6f_MYH7_loc_sum_vs_linear_model_org_by_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_6f_MYH7_loc_sum_vs_linear_model_org_by_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Speculative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (4, 4)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "g = sns.scatterplot(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x='Cell area (μm^2)',\n",
    "    y=\"FISH probe count\",\n",
    "    hue=\"FISH probe\",\n",
    "    s=3,\n",
    "    hue_order=PROBE_ORDER,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    ")\n",
    "g.set(xlim=(0,10_000));\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.2), frameon=False)\n",
    "sns.despine();\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_supp_cell_area_vs_probe_count.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_supp_cell_area_vs_probe_count.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.FacetGrid(\n",
    "    data=df_tidy.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    col=\"FISH probe\",\n",
    "    col_order=PROBE_ORDER,\n",
    "    col_wrap=4,\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    sharex=True,\n",
    "    xlim=(0,10_000),\n",
    "    sharey=False,\n",
    "    height=3.0,\n",
    "    aspect=1.0,\n",
    ")\n",
    "\n",
    "g = g.map(\n",
    "    plt.scatter,\n",
    "    'Cell area (μm^2)',\n",
    "    \"FISH probe count\",\n",
    "    s=3,\n",
    "    alpha=0.5,\n",
    "    linewidth=0,\n",
    ").add_legend()\n",
    "\n",
    "g.set_titles(\"{col_name}\")\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_supp_cell_area_vs_probe_count_faceted.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_supp_cell_area_vs_probe_count_faceted.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "boot_spearmanr(\n",
    "    df_tidy,\n",
    "    x='Cell area (μm^2)',\n",
    "    y=\"FISH probe count\",\n",
    "    stratify=\"FISH probe\",\n",
    "    N=1000\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "age = 18\n",
    "\n",
    "df_area_corr = boot_spearmanr(\n",
    "    df_tidy[df_tidy[\"Cell age\"] == age],\n",
    "    x='Cell area (μm^2)',\n",
    "    y=\"FISH probe count\",\n",
    "    stratify=\"FISH probe\",\n",
    "    N=1000\n",
    ")\n",
    "\n",
    "df_area_corr = pd.concat([pd.DataFrame({\"Cell age\":[age]*len(df_area_corr)}), df_area_corr], axis=1)\n",
    "df_area_corr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "age = 32\n",
    "\n",
    "df_area_corr = boot_spearmanr(\n",
    "    df_tidy[df_tidy[\"Cell age\"] == age],\n",
    "    x='Cell area (μm^2)',\n",
    "    y=\"FISH probe count\",\n",
    "    stratify=\"FISH probe\",\n",
    "    N=1000\n",
    ")\n",
    "\n",
    "df_area_corr = pd.concat([pd.DataFrame({\"Cell age\":[age]*len(df_area_corr)}), df_area_corr], axis=1)\n",
    "df_area_corr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dims = (4,3)\n",
    "fig, ax = plt.subplots(figsize=dims)\n",
    "\n",
    "ax = sns.violinplot(\n",
    "    data=df_org_scores_all,\n",
    "    x=\"Expert structural annotation score (all)\",\n",
    "    y=\"Combined organizational score\",\n",
    "    hue=\"Cell age\",\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    split=True,\n",
    "    linewidth=1\n",
    ")\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.0, 0.7), frameon=False, title=\"Cell age\");\n",
    "sns.despine()\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/'fig_supp_org_score_hist_by_expert_score_and_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/'fig_supp_org_score_hist_by_expert_score_and_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "targets = [\n",
    "    'HPRT1 (count/μm^2)',\n",
    "    'COL2A1 (count/μm^2)',\n",
    "    'H19 (count/μm^2)',\n",
    "    'ATP2A2 (count/μm^2)',\n",
    "    'MYH6 (count/μm^2)',\n",
    "    'MYH7 (count/μm^2)',\n",
    "    'BAG3 (count/μm^2)',\n",
    "    'TCAP (count/μm^2)',\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for target in targets:\n",
    "    \n",
    "    # bootstrapped regression\n",
    "    df_boot_reg = boot_regression(\n",
    "        df.dropna(subset=[target]).reset_index(drop=True),\n",
    "        y_col=target,\n",
    "        weight_col=None,\n",
    "        N=1000\n",
    "    )\n",
    "    \n",
    "    # get conf intervals from bootstrapped outputs\n",
    "    reg_plot_ci_df = make_reg_plot_ci_df(df_boot_reg)\n",
    "    \n",
    "    # make the featue importance plot\n",
    "    fig_bar, ax_bar = make_regression_bar_plot(reg_plot_ci_df, title=target)\n",
    "    \n",
    "    # save png and svg\n",
    "    if SAVE:\n",
    "        plt.savefig(regression_sup_dir_pngs/f'fig_supp_linear_model_feature_importances_{target.split()[0]}.png', dpi=100, bbox_inches = \"tight\")\n",
    "        plt.savefig(regression_sup_dir_svgs/f'fig_supp_linear_model_feature_importances_{target.split()[0]}.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for target in targets:\n",
    "    \n",
    "    df_pred_true  = get_pred_true(\n",
    "        df.dropna(subset=[target]).reset_index(drop=True),\n",
    "        y_col=target,\n",
    "        weight_col=None,\n",
    "        meta_cols=[\"Cell age\"]\n",
    "    )\n",
    "    \n",
    "    spr = np.mean(\n",
    "        [\n",
    "            spearmanr(\n",
    "                df_pred_true[df_pred_true[\"Cell age\"]==day][target],\n",
    "                df_pred_true[df_pred_true[\"Cell age\"]==day][f'{target} predicted'])[0]\n",
    "            for day in [18, 32]\n",
    "        ]\n",
    "    )\n",
    "\n",
    "    title = f'{target.replace(\"_\", \" \")}\\nMean Spearman R = {spr:.2f}'\n",
    "    \n",
    "    fig_scatter, ax_scatter = make_regression_scatter_plot(df_pred_true, target=target, title=title);\n",
    "\n",
    "    # save png and svg\n",
    "    if SAVE:\n",
    "        plt.savefig(regression_sup_dir_pngs/f'fig_supp_linear_model_pred_acc_{target.split()[0]}.png', dpi=100, bbox_inches = \"tight\")\n",
    "        plt.savefig(regression_sup_dir_svgs/f'fig_supp_linear_model_pred_acc_{target.split()[0]}.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.pairplot(\n",
    "    df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    y_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    hue='Cell age',\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    plot_kws=dict(\n",
    "        linewidth=0,\n",
    "        alpha = 0.25,\n",
    "        s=20\n",
    "    ),\n",
    "    diag_kws=dict(\n",
    "        bw=0.025\n",
    "    )\n",
    ");\n",
    "\n",
    "mcv_lims = (0,1)\n",
    "ph_lims = (0.15,0.85)\n",
    "pd_lims = (0.5,3.0)\n",
    "\n",
    "g.axes[0,0].set_xlim(mcv_lims)\n",
    "g.axes[1,1].set_xlim(ph_lims)\n",
    "g.axes[2,2].set_xlim(pd_lims)\n",
    "\n",
    "g.axes[0,1].set_xlim(ph_lims)\n",
    "g.axes[0,1].set_ylim(mcv_lims)\n",
    "g.axes[1,0].set_xlim(mcv_lims)\n",
    "g.axes[1,0].set_ylim(ph_lims)\n",
    "\n",
    "g.axes[0,2].set_xlim(pd_lims)\n",
    "g.axes[0,2].set_ylim(mcv_lims)\n",
    "g.axes[2,0].set_xlim(mcv_lims)\n",
    "g.axes[2,0].set_ylim(pd_lims)\n",
    "\n",
    "g.axes[1,2].set_xlim(pd_lims)\n",
    "g.axes[1,2].set_ylim(ph_lims)\n",
    "g.axes[2,1].set_xlim(ph_lims)\n",
    "g.axes[2,1].set_ylim(pd_lims);\n",
    "\n",
    "plt.subplots_adjust(hspace=0.2, wspace=0.2)\n",
    "g._legend.set_bbox_to_anchor((1.05, 0.5))\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_supp_global_alignment_correlations_day.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_supp_global_alignment_correlations_day.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.pairplot(\n",
    "    df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    y_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    hue=\"Expert structural annotation score (grouped)\",\n",
    "    hue_order=[\"1-2\", \"3\", \"4-5\"],\n",
    "    aspect=1,\n",
    "    plot_kws=dict(\n",
    "        linewidth=0,\n",
    "        alpha = 0.25,\n",
    "        s=20\n",
    "    ),\n",
    "    diag_kws=dict(\n",
    "        bw=0.025\n",
    "    )\n",
    ");\n",
    "\n",
    "mcv_lims = (0,1)\n",
    "ph_lims = (0.15,0.85)\n",
    "pd_lims = (0.5,3.0)\n",
    "\n",
    "g.axes[0,0].set_xlim(mcv_lims)\n",
    "g.axes[1,1].set_xlim(ph_lims)\n",
    "g.axes[2,2].set_xlim(pd_lims)\n",
    "\n",
    "g.axes[0,1].set_xlim(ph_lims)\n",
    "g.axes[0,1].set_ylim(mcv_lims)\n",
    "g.axes[1,0].set_xlim(mcv_lims)\n",
    "g.axes[1,0].set_ylim(ph_lims)\n",
    "\n",
    "g.axes[0,2].set_xlim(pd_lims)\n",
    "g.axes[0,2].set_ylim(mcv_lims)\n",
    "g.axes[2,0].set_xlim(mcv_lims)\n",
    "g.axes[2,0].set_ylim(pd_lims)\n",
    "\n",
    "g.axes[1,2].set_xlim(pd_lims)\n",
    "g.axes[1,2].set_ylim(ph_lims)\n",
    "g.axes[2,1].set_xlim(ph_lims)\n",
    "g.axes[2,1].set_ylim(pd_lims);\n",
    "\n",
    "plt.subplots_adjust(hspace=0.2, wspace=0.2)\n",
    "g._legend.set_bbox_to_anchor((0.8, 0.5))\n",
    "g._legend.set_title(\"Expert score  \")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_supp_global_alignment_correlations_score.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_supp_global_alignment_correlations_score.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.pairplot(\n",
    "    df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    y_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    hue='Cell age',\n",
    "    hue_order=[18, 32],\n",
    "    palette=DAY_COLOR_PALETTE,\n",
    "    plot_kws=dict(\n",
    "        linewidth=0,\n",
    "        alpha = 0.25,\n",
    "        s=20\n",
    "    ),\n",
    "    diag_kws=dict(\n",
    "        bw=0.1\n",
    "    )\n",
    ");\n",
    "\n",
    "mcv_lims = (0,1)\n",
    "ph_lims = (0.15,0.85)\n",
    "pd_lims = (0.5,3.0)\n",
    "\n",
    "g.axes[0,0].set_xlim(mcv_lims)\n",
    "g.axes[1,1].set_xlim(ph_lims)\n",
    "g.axes[2,2].set_xlim(pd_lims)\n",
    "\n",
    "g.axes[0,1].set_xlim(ph_lims)\n",
    "g.axes[0,1].set_ylim(mcv_lims)\n",
    "g.axes[1,0].set_xlim(mcv_lims)\n",
    "g.axes[1,0].set_ylim(ph_lims)\n",
    "\n",
    "g.axes[0,2].set_xlim(pd_lims)\n",
    "g.axes[0,2].set_ylim(mcv_lims)\n",
    "g.axes[2,0].set_xlim(mcv_lims)\n",
    "g.axes[2,0].set_ylim(pd_lims)\n",
    "\n",
    "g.axes[1,2].set_xlim(pd_lims)\n",
    "g.axes[1,2].set_ylim(ph_lims)\n",
    "g.axes[2,1].set_xlim(ph_lims)\n",
    "g.axes[2,1].set_ylim(pd_lims);\n",
    "\n",
    "plt.subplots_adjust(hspace=0.2, wspace=0.2)\n",
    "g._legend.set_bbox_to_anchor((1.05, 0.5))\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_supp_global_alignment_correlations_day_smoother.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_supp_global_alignment_correlations_day_smoother.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.pairplot(\n",
    "    df.sample(frac=1, replace=False).reset_index(drop=True),\n",
    "    x_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    y_vars=['Max coefficient var', 'Peak height', 'Peak distance (μm)'],\n",
    "    hue=\"Expert structural annotation score (grouped)\",\n",
    "    hue_order=[\"1-2\", \"3\", \"4-5\"],\n",
    "    aspect=1,\n",
    "    plot_kws=dict(\n",
    "        linewidth=0,\n",
    "        alpha = 0.25,\n",
    "        s=20\n",
    "    ),\n",
    "    diag_kws=dict(\n",
    "        bw=0.1\n",
    "    )\n",
    ");\n",
    "\n",
    "mcv_lims = (0,1)\n",
    "ph_lims = (0.15,0.85)\n",
    "pd_lims = (0.5,3.0)\n",
    "\n",
    "g.axes[0,0].set_xlim(mcv_lims)\n",
    "g.axes[1,1].set_xlim(ph_lims)\n",
    "g.axes[2,2].set_xlim(pd_lims)\n",
    "\n",
    "g.axes[0,1].set_xlim(ph_lims)\n",
    "g.axes[0,1].set_ylim(mcv_lims)\n",
    "g.axes[1,0].set_xlim(mcv_lims)\n",
    "g.axes[1,0].set_ylim(ph_lims)\n",
    "\n",
    "g.axes[0,2].set_xlim(pd_lims)\n",
    "g.axes[0,2].set_ylim(mcv_lims)\n",
    "g.axes[2,0].set_xlim(mcv_lims)\n",
    "g.axes[2,0].set_ylim(pd_lims)\n",
    "\n",
    "g.axes[1,2].set_xlim(pd_lims)\n",
    "g.axes[1,2].set_ylim(ph_lims)\n",
    "g.axes[2,1].set_xlim(ph_lims)\n",
    "g.axes[2,1].set_ylim(pd_lims);\n",
    "\n",
    "plt.subplots_adjust(hspace=0.2, wspace=0.2)\n",
    "g._legend.set_bbox_to_anchor((0.8, 0.5))\n",
    "g._legend.set_title(\"Expert score  \")\n",
    "\n",
    "# save png and svg\n",
    "if SAVE:\n",
    "    plt.savefig(save_dir_pngs/f'fig_supp_global_alignment_correlations_score_smoother.png', dpi=100, bbox_inches = \"tight\")\n",
    "    plt.savefig(save_dir_svgs/f'fig_supp_global_alignment_correlations_score_smoother.svg', format=\"svg\", bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:rory_cfd]",
   "language": "python",
   "name": "conda-env-rory_cfd-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
